core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -158,11 +157,6 @@
     private static int queryNameId = 0;
     private static SearchModule searchModule;
 
-
-    protected static QueryShardContext queryShardContext() {
-        return queryShardContext;
```

Comment:
Okay, I think I get it. I went ahead and removed the queryShardContext instance entirely. I had to introduce a few more static fields for all the services we need in init(), but I think it is better this way. Also pulled up some mocking on to of the SearchContext that was done in the Parent/Child/Nested tests but that can be used everywhere I think.